packages/perseus/src/widgets/label-image/label-image.tsx

```diff
@@ -67,23 +68,14 @@ type Point = {
     y: number;
 };
 
-// TODO: should this be using WidgetProps / PerseusLabelImageWidgetOptions?
 type LabelImageProps = ChangeableProps &
-    DependencyProps & {
+    DependencyProps &
+    // TODO: there's some weirdness in our types between
+    // PerseusLabelImageMarker and InteractiveMarkerType
```

Comment:
I'm pretty sure the weirdness is that this component triggers `onChange` callbacks and "enhances" the items in the `markers` with what is essentially state. The `Renderer` then renders this widget with these enhanced `markers`. On first render, they will be of type `PerseusLabelImageMarker`, but after the first `onChange` they will be `InteractiveMarkerType` objects (which just add a few keys to the base `PerseusLabelImageMarker`).

Reply:
I think this is going to get addressed in https://khanacademy.atlassian.net/browse/LEMS-2440